### DISCUSSION TITLE
DOM - Desenvolvimento orientado a modinha/ Hype Driven Development

### DISCUSSION THREAD

**Discussion by flpms** [frontendbr/forum#561] [Perguntas e Respostas]
Linguagens, frameworks e etc, são ferramentas. Se ao dizer que martelo é excelente em colocar pregos e lógicamente não usamos ele em tudo, por que as vezes fazemos o mesmo com as ferramentas de desenvolvimento e programação?

**Comment by kivervinicius** [comment]
Mas o martelo tem cabo emborrachado? Tem porta pregos na lateral? Com uma marreta teria a mesma utilidade? Até nas pequenas ferramentas temos benefícios e custos diferentes.

**Comment by flpms** [comment]
@kivervinicius a questão não são as _features_ do martelo. Mas querer usar ele para fazer todos os tipos de trabalho e no desenvolvimento orientado a modinha é isso que acontece.

**Comment by munizart** [comment]
Como já dizia [nosso camarada:](https://www.youtube.com/watch?v=SmwngbNORDY&feature=youtu.be&t=61)
> Sendo uma ferramenta cada um tem um propósito diferente.

Brincadeiras (ou nem tanto) a parte, acho que ferramentas nascem da necessidade de resolver um problema, e portanto, nascem para resolver aquele problema.

>Se ao dizer que martelo é excelente em colocar pregos e lógicamente não usamos ele em tudo, por que as vezes fazemos o mesmo com as ferramentas de desenvolvimento e programação?

Exatamente por isso. Assim como o martelo serve para martelar, cada ferramenta tem seu propósito.

**Comment by eduardojmatos** [comment]
Bem bacana o post e a discussão! Gravei com o pessoal do DevNaEstrada, @femontanha, @raymonsanches e Vinícius um episódio sobre isso: http://devnaestrada.com.br/2017/04/28/dom.html!

**Comment by gabrieldarezzo** [comment]
Mas e ai @eduardojmatos @femontanha @raymonsanches o Front-end está morrendo kkkkk?


Ps, sou fã do PodCast kkkk

**Comment by raymonsanches** [comment]
hAUEhuahe... quem sou pra falar uma coisa dessas. Sou só um backend ;)

**Comment by eduardojmatos** [comment]
Melhor não entrar nesse assunto heheh
